tvx/messages_message_application.go

```diff
@@ -52,26 +52,25 @@ func MessageTest_MessageApplicationEdgecases() error {
 	}
 
 	err = func(testname string) error {
-		//TODO: this test is broken, fix later
-		return nil
 		td := drivers.NewTestDriver()
 
 		v := newEmptyMessageVector()
-
 		alice, _ := td.NewAccountActor(drivers.SECP, aliceBal)
 
 		preroot := td.GetStateRoot()
 		msg := td.MessageProducer.Transfer(alice, alice, chain.Value(transferAmnt), chain.Nonce(0), chain.GasPrice(10), chain.GasLimit(1))
+		v.ApplyMessages = append(v.ApplyMessages, chain.MustSerialize(msg))
+
 		// Expect Message application to fail due to lack of gas
 		td.ApplyFailure(
 			msg,
 			exitcode_spec.SysErrOutOfGas)
 
-		// Expect Message application to fail due to lack of gas when sender is unknown
 		unknown := chain.MustNewIDAddr(10000000)
 		msg = td.MessageProducer.Transfer(unknown, alice, chain.Value(transferAmnt), chain.Nonce(0), chain.GasPrice(10), chain.GasLimit(1))
```

Comment:
I am not 100% sure I understand this test. How can the sender be unknown? Isn't this messages supposed to be stopped on the mempool level, and never reach the VM?

Reply:
cc @frrist (marking you as I see you might be the original author of this test and shed some light ;)

Reply:
Based on chats we've had today, it appears that such a message really should not be part of a `valid` block, but it is always good to confirm that the VMs of various clients would end up at the same state, if we throw such an invalid message to the VM.

Reply:
In the event that an invalid message (like the one here) isn't filtered by block validation we need to ensure behavior is consistent across VM implementations. The assertion being made here is checking the order of validation checks in a filecoin VM:
[Lotus Validation order](https://github.com/filecoin-project/lotus/blob/3058f280d9ba34c044055e2a85a63bd07d52b307/chain/vm/vm.go#L329-L345)
[go-filecoin Validation Order](https://github.com/filecoin-project/go-filecoin/blob/8f6abbaaec2df92eb9a9ededbfc1d9dee035c987/internal/pkg/vm/internal/vmcontext/vmcontext.go#L374-L394)

Reply:
Yeah, these kinds of messages would really benefit from capturing the `SysErrOutOfGas` in the new postconditions receipts.